osu.Game/Screens/Select/FooterButtonRandom.cs

```diff
@@ -67,5 +55,35 @@ private void updateText()
                 secondaryText.FadeOut(120, Easing.InQuad);
             }
         }
+
+        public override bool OnPressed(GlobalAction action)
+        {
+            switch (action)
+            {
+                case GlobalAction.SelectPreviousRandom:
+                    rewindSearch = true;
+                    Action = PreviousRandom;
+                    updateText();
+                    Click();
+                    return true;
+
+                case GlobalAction.SelectNextRandom:
+                    Action = NextRandom;
+                    updateText();
+                    Click();
+                    return true;
+            }
+
+            return false;
```

Comment:
I'm not sure on this piece of code, it feels like it could be improved upon.

* The `Action` substitution feels weird. You could just do this in the ctor:

```csharp
Action = () =>
{
    if (rewindSearch)
        PreviousRandom?.Invoke();
    else
        NextRandom?.Invoke();
};
```

* After that, this switch can be rewritten as:

```csharp
switch (action)
{
    case GlobalAction.SelectPreviousRandom:
        rewindSearch = true;
        break;

    case GlobalAction.SelectNextRandom:
        break;

    default:
        return false;
}

updateText();
Click();
return true;
```

Not sure how much better that is. Feels marginally better to me but I can see differing opinions on that.